pynamodb/models.py

```diff
@@ -324,15 +335,42 @@ def __repr__(self):
                 msg = "{}<{}>".format(self.Meta.table_name, serialized.get(HASH))
             return six.u(msg)
 
-    def delete(self, condition=None):
+    @classmethod
+    def condition_check(cls, hash_key, in_transaction, condition, range_key=None):
+        hash_key, range_key = cls._serialize_keys(hash_key, range_key)
+        operation_kwargs = cls._get_connection().get_operation_kwargs_for_condition_check(
+            hash_key=hash_key,
+            range_key=range_key,
+            condition=condition,
+        )
+        in_transaction.add_condition_check_item(
+            model_cls=cls,
+            hash_key=hash_key,
+            range_key=range_key,
+            operation_kwargs=operation_kwargs
+        )
+
+    def delete(self, condition=None, in_transaction=None):
         """
         Deletes this object from dynamodb
         """
         args, kwargs = self._get_save_args(attributes=False, null_check=False)
         kwargs.update(condition=condition)
+
+        if in_transaction is not None:
+            operation_kwargs = self._get_connection().get_operation_kwargs_for_delete_item(*args, **kwargs)
+            in_transaction.add_delete_item(model=self, operation_kwargs=operation_kwargs)
+            return {}
         return self._get_connection().delete_item(*args, **kwargs)
 
-    def update(self, actions, condition=None):
+    def _update_item_with_raw_data(self, data):
```

Comment:
@jpinner-lyft does this mesh with your work in #644?

Reply:
i can actually undo making this function because im not using it the way i originally intended to